project_task_description_template/tests/test_task_template.py

```diff
@@ -0,0 +1,183 @@
+        cls.tag_yellow = tag_model.create({"name": "Yellow"})
+        # create res users model
+        res_users_model = cls.env["res.users"]
+        # create users: Bob, Mike, Kate
+        cls.user_bob = res_users_model.create(
+            {"name": "Bob", "email": "bob@example.com", "login": "bob"}
+        )
+        cls.user_mike = res_users_model.create(
+            {"name": "Mike", "email": "mike@example.com", "login": "mike"}
+        )
+        cls.user_kate = res_users_model.create(
+            {"name": "Kate", "email": "kate@example.com", "login": "kate"}
+        )
+        # create task template model
+        task_template_model = cls.env["project.task.template"]
+        # create task templates:
+
+        cls.template_1 = task_template_model.create(
+            {
+                "name": "Template1",
+                "tag_ids": [
+                    cls.tag_red.id,
+                    cls.tag_green.id,
+                ],
+                "user_id": cls.user_bob.id,
+                "description": "This is template 1",
+            }
+        )
+        cls.template_2 = task_template_model.create(
+            {
+                "name": "Template2",
+                "tag_ids": [
+                    cls.tag_blue.id,
+                    cls.tag_white.id,
+                ],
+                "user_id": cls.user_mike.id,
+                "description": "This is template 2",
+            }
+        )
+        # create project model
+        project_model = cls.env["project.project"]
+        # create projects : "Project 1" and add templates 1-2
+        cls.project_1 = project_model.create(
+            {
+                "name": "Project 1",
+                "task_template_ids": [cls.template_1.id, cls.template_2.id],
+            }
+        )
+        # create task model
+        project_task_model = cls.env["project.task"]
+        cls.test_task_1 = project_task_model.create(
+            {
+                "name": "Test task 1",
+                "user_id": cls.user_kate.id,
+                "tag_ids": [
+                    cls.tag_yellow.id,
+                ],
+                "description": "This is task1",
+                "task_template_id": False,
+            }
+        )
+
+    def test_task_1_template_1(self):
+        """Test the behavior when choosing a template."""
+        # Set the task_template_id of test_task_1 to template_1.id
+        self.test_task_1.task_template_id = self.template_1.id
+
+        # Trigger the _onchange_task_template_id method
+        self.test_task_1._onchange_task_template_id()
+
+        # Verify that the user_id of test_task_1 is equal to the user_id of template_1
+        self.assertEqual(
+            self.test_task_1.user_id.id,
+            self.template_1.user_id.id,
+            msg="The task 1 user #id must be equal to the template 1 user ID#{user_id}".format(
+                user_id=self.template_1.user_id.id
+            ),
```

Comment:
**nitpick(non-blocking, minor):** Please use `%` instead of `format()`.

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#idioms

Reply:
Thanks, I'll fix it